amundsen_application/static/js/ducks/issue/tests/index.spec.ts

```diff
@@ -0,0 +1,241 @@
+      const stateIssues: Issue[]=[];
+      remaining = 0; 
+      remainingUrl = 'testUrl'; 
+      testState = { 
+        isLoading: false, 
+        issues: stateIssues, 
+        remainingIssues: remaining, 
+        remainingIssuesUrl: remainingUrl
+      };
+     
+    });
+
+    it('should return the existing state if action is not handled', () => {
+      expect(reducer(testState, { type: 'INVALID.ACTION' })).toEqual(testState);
+    });
+
+    it('should handle GetIssues.REQUEST', () => {
+      expect(reducer(testState, getIssues(tableKey))).toEqual({ 
+        issues: [], 
+        isLoading: true, 
+        remainingIssuesUrl: null, 
+        remainingIssues: 0
+      });
+    });
+
+    it('should handle GetIssues.SUCCESS', () => {
+      expect(reducer(testState, getIssuesSuccess(issues, remaining, remainingUrl))).toEqual({ 
+        issues, 
+        isLoading: false,
+        remainingIssues: remaining, 
+        remainingIssuesUrl: remainingUrl
+      });
+    });
+
+    it('should handle GetIssues.FAILURE', () => {
+      expect(reducer(testState, getIssuesFailure())).toEqual({ 
+        issues: [], 
+        isLoading: false, 
+        remainingIssuesUrl: null,
+        remainingIssues: remaining 
+      });
+    });
+
+    it('should handle CreateIssue.REQUEST', () => {
+      expect(reducer(testState, createIssue(formData))).toEqual({ 
+        issues: [], 
+        isLoading: true, 
+        remainingIssuesUrl: remainingUrl,
+        remainingIssues: remaining 
+       });
+    });
+
+    it('should handle CreateIssue.SUCCESS', () => {
+      expect(reducer(testState, createIssueSuccess(issue))).toEqual({
+         ...testState, issues: [issue], isLoading: false });
+    });
+
+    it('should handle CreateIssue.FAILURE', () => {
+      expect(reducer(testState, createIssueFailure())).toEqual({ issues: [], 
+        isLoading: false, 
+        remainingIssuesUrl: remainingUrl,
+        remainingIssues: remaining 
+      });
+    });
+  });
+
+  describe('sagas', () => {
+    describe('getIssuesWatcher', () => {
+      it('takes every getIssues.REQUEST with getIssuesWatcher', () => {
+        testSaga(getIssuesWatcher)
+          .next().takeEvery(GetIssues.REQUEST, getIssuesWorker)
+          .next().isDone();
+      });
+    });
+
+    describe('getIssuesWorker', () => {
+      let action: GetIssuesRequest;
+      let remainingIssuesUrl: string;
+      let remainingIssues: number; 
+      beforeAll(() => {
+        action = getIssues(tableKey);
+        issues = globalState.issue.issues;
+        remainingIssues = globalState.issue.remainingIssues; 
+        remainingIssuesUrl = globalState.issue.remainingIssuesUrl;
+      });
+
+      it('gets issues', () => {
+        return expectSaga(getIssuesWorker, action)
+          .provide([
+            [matchers.call.fn(API.getIssues), {issues, remainingIssues, remainingIssuesUrl}],
+          ])
+          .put(getIssuesSuccess(issues, undefined, undefined))
```

Comment:
Rather than putting `undefined, undefined` you can add a `?` to the function definition so they become optional parameters.

`getIssuesSuccess(issues: Issue[], remainingIssues?: number, remainingIssuesUrl?: string)`

Reply:
ty!